src/manager/evm/manager.rs

```diff
@@ -78,6 +80,36 @@ struct IPCContractInfo {
     provider: Provider<Http>,
 }
 
+#[async_trait]
+impl TopDownCheckpointQuery for EthSubnetManager {
+    async fn chain_head(&self) -> Result<ChainHeadResponse> {
+        unimplemented!()
```

Comment:
Are you planning to implemented in the future, why is this unimplemented? (add a new issue if you are planning a follow-up and ideally point it from here). Thanks!

Reply:
It's kind of difficult to do as Evm does not necessarily have all the types in `ChainHeadResponse`. Come to think of it, it does not make sense to have this trait implementation. I have removed this implementation.

Reply:
It's kind of difficult to do as Evm does not necessarily have all the types in `ChainHeadResponse`. Come to think of it, it does not make sense to have this trait implementation. I have updated the trait definition to return chain epoch as this is what top down checkpoint in fendermint needs.